testclock/dilated.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2021 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package testclock
+
+import (
+	"time"
+
+	"github.com/juju/clock"
+)
+
+// NewDilatedWallClock returns a clock that can be sped up or slowed down.
+// dilationFaction is the real duration of a second.
+func NewDilatedWallClock(dilationFactor time.Duration) clock.Clock {
```

Comment:
I saw `dilationFactor` and was expecting a scalar multiplier. ie a factor of 2 would slow down time and 0.5 would speed it up.
If we want to pass in the duration of a real second instead, could we rename the underlying attribute, eg `realSecondDuration` or something